### ISSUE TITLE
issue in ApplicationManifestUtils.write method 

### ISSUE THREAD

**Issue by parasuramln** [cloudfoundry/cf-java-client#911]
`ApplicationManifestUtils.write(OutputStream out, List<ApplicationManifest> applicationManifests)` **1.giving manifest in which memory does not have any unit of measurement.like M, MB, G, or GB.
2.path in output manifest is AbsolutePath .(expected is whatever available in old manifest).**

i used the following manifest to ```ApplicationManifestUtils.read```:
```
---
applications:
- name: apicomp
  memory: 1024M
  buildpack: java_buildpack
  instances: 1
  path: /Backend/demo-0.0.1-SNAPSHOT.jar
  random-route: true 
  services:
    - autodeployinst  
- name: uicomp
  memory: 1024M
  buildpack: staticfile_buildpack
  instances: 1
  path: /UI_Codes/dist/
  random-route: true 
```
then used
`ApplicationManifestUtils.write`  to  modify appName and got manifest output as:

```
---
applications:
- buildpack: java_buildpack
  instances: 1
  memory: 1024
  name: apicomp-green
  path: C:\Backend\demo-0.0.1-SNAPSHOT.jar
  random-route: true
  services:
  - autodeployinst
- buildpack: staticfile_buildpack
  instances: 1
  memory: 1024
  name: uicomp-green
  path: C:\UI_Codes\dist
  random-route: true
```

**Comment by nebhale** [comment]
In a manifest, memory without a suffix is interpreted as megabytes.  This means that `1024M` and `1024` are equivalent to the API.  In fact, under the hood the REST API that backs creation of applications [*only* accepts an integer of megabytes](https://apidocs.cloudfoundry.org/4.4.0/apps/creating_an_app.html).

`ApplicationManifestUtils` does not aim to preserve the input values and ordering when outputting, instead writing a _functionally equivalent_ manifest.

**Comment by parasuramln** [comment]
but without any measurement tries to push app ,it is giving following error.

**Byte quantity must be a positive integer with a unit of measurement like M, MB, G, or GB**
**FAILED**

**Comment by nebhale** [comment]
I don't [see that string in the Java Client](https://github.com/cloudfoundry/cf-java-client/search?q=Byte+quantity&unscoped_q=Byte+quantity).  What's the stack trace for the error?

**Comment by parasuramln** [comment]
The above error I got from CLI

**Comment by parasuramln** [comment]
Use case is:

Read blue version app manifest ,
Create new manifest from it by modifying required parameters like appname..

**Comment by nebhale** [comment]
Yeah, this definitely a bug.  While the underlying implementations are all doing the right thing with an integer of megabytes, the [spec](https://docs.cloudfoundry.org/devguide/deploy-apps/manifest.html#memory) clearly states that a letter must be present.  @twoseat to fix this I think you’ll need to write a custom serializer for this that appends an `M` after the number is written.

**Comment by parasuramln** [comment]
2.path in output manifest is AbsolutePath .(expected is whatever available in old manifest).

**Comment by parasuramln** [comment]
> I forgot the appropriate tag, but these should be resolved in [1660e86](https://github.com/cloudfoundry/cf-java-client/commit/1660e86a3eca702edc80cc1dbd6ea5de2880902b)

hi 
corresponding POJO **ApplicationManifest** changes are not there in this commit.
please change
`private final @Nullable Integer memory;` to `private final @Nullable String memory`;
and
`private final @Nullable Integer disk`; to` private final @Nullable String disk`;
